psydac/fem/projectors.py

```diff
@@ -158,3 +160,335 @@ def get_dual_dofs(Vh, f, domain_h, backend_language="python", return_format='ste
+                bcs += [ebcs[2], ebcs[3]] if periodic[1] == False else []
+            if dim == 3:
+                bcs += [ebcs[4], ebcs[5]] if periodic[2] == False else []
+
+        elif kind == 'hcurl':
+            assert dim in (2, 3)
+            bcs_x = [ebcs[2], ebcs[3]] if periodic[1] == False else []
+            if dim == 3:
+                bcs_x += [ebcs[4], ebcs[5]] if periodic[2] == False else []
+            bcs_y = [ebcs[0], ebcs[1]] if periodic[0] == False else []
+            if dim == 3:
+                bcs_y += [ebcs[4], ebcs[5]] if periodic[2] == False else []
+            if dim == 3:
+                bcs_z = [ebcs[0], ebcs[1]] if periodic[0] == False else []
+                bcs_z += [ebcs[2], ebcs[3]] if periodic[1] == False else []
+            bcs = [bcs_x, bcs_y]
+            if dim == 3:
+                bcs.append(bcs_z)
+
+        elif kind == 'hdiv':
+            assert dim in (2, 3)
+            bcs_x = [ebcs[0], ebcs[1]] if periodic[0] == False else []
+            bcs_y = [ebcs[2], ebcs[3]] if periodic[1] == False else []
+            if dim == 3:
+                bcs_z = [ebcs[4], ebcs[5]] if periodic[2] == False else []
+            bcs = [bcs_x, bcs_y]
+            if dim == 3:
+                bcs.append(bcs_z)
+
+        else:
+            raise ValueError(f'{kind} must be either "h1", "hcurl" or "hdiv"')
+        
+        return tuple(bcs)
+
+#===============================================================================
+class MultipatchDirichletProjector(DirichletProjector):
+    """
+    A LinearOperator (for multipatch domains) that applies homogeneous (unless manually given different bcs) Dirichlet boundary conditions.
+
+    Parameters
+    ----------
+    fem_space : psydac.fem.basic.FemSpace
+        fem_space.coeff_space is domain and codomain of this LO. fem_space.kind.name determines the BCs that get applied.
+
+    bcs : Iterable | None
+        Iterable of sympde.topology.Boundary objects. 
+        Allows the user to apply different kinds of BCs.
+        Must not be passed if a space_kind argument is passed.
+
+    space_kind : str | SpaceType | None
+        Necessary only if fem_space.kind.name is undefined and no bcs are passed.
+        Must not be passed if a space_kind argument is passed.
+
+    Notes
+    -----
+    See examples/vector_potential_3d.py for a use case of such an operator.
+    
+    """
+    def __init__(self, fem_space, *, bcs=None, space_kind=None):
+
+        assert isinstance(fem_space, FemSpace)
+        assert fem_space.is_multipatch
+        assert bcs is None or isinstance(bcs, Iterable)
+        assert space_kind is None or isinstance(space_kind, (str, SpaceType))
+        if bcs is not None:
+            assert space_kind is None
+        if space_kind is not None:
+            assert bcs is None
+
+        dim = fem_space.symbolic_space.domain.dim
+        assert dim == 2, 'The class MultipatchBoundaryProjector is implemented only in 2D.'
```

Comment:
The class name in the error message is wrong, sorry for the confusion! 😅

We can make sure to print the correct class name by using `__class__`:
```suggestion
        assert dim == 2, f'The class {__class__.__name__} is implemented only in 2D.'
```